packages/rich-text/src/plugins/Editor/index.ts

```diff
@@ -0,0 +1,33 @@
+import { Editor, Element, Text } from 'slate';
+import { CustomEditor, CustomElement } from '../../types';
+
+export function withEditorPlugin(editor: CustomEditor): CustomEditor {
+  const { isVoid: originalIsVoid } = editor;
+
+  function isBlockSelected(type: string): boolean {
+    const [match] = Array.from(
+      Editor.nodes(editor, {
+        match: (node) => Element.isElement(node) && (node as CustomElement).type === type,
+      })
+    );
```

Comment:
This seems kinda odd to me but maybe I just don't understand [the API](https://docs.slatejs.org/api/nodes/editor#editor-nodes-editor-editor-options-generator-less-than-nodeentry-less-than-t-greater-than-void-undefined-greater-than). I would guess from this logic this doesn't tell us whether the block is selected, but whether a block of type `type` appears in the editor markup somewhere. Is that right? And where does this get used?

Reply:
I noticed from the small experience that I had that it [always gets from the current selection](https://docs.slatejs.org/concepts/04-transforms#the-at-option) unless we manually override it. I think I forgot to remove it, but I used it at some point, however, I'm using the same thing for the headings.